packages/driver/test/cypress/integration/cypress/cypress_spec.coffee

```diff
@@ -47,6 +47,18 @@ describe "driver/src/cypress/index", ->
 
         done()
 
+    ## https://github.com/cypress-io/cypress/issues/4346
+    it "can complete if a circular reference is sent", ->
+      foo = {
+        bar: {}
+      }
+
+      foo.bar.baz = foo
+
+      Cypress.backend("foo", foo)
```

Comment:
You need to add the `.then -> throw new Error(...)` bits here so that we know the `.catch()` hit